src/internal_module/fs.rs

```diff
@@ -579,45 +586,46 @@ fn open_sync(ctx: &mut Context, _this_val: JsValue, arg: &[JsValue]) -> JsValue
         if let Some(JsValue::Int(flag)) = arg.get(1) {
             if let Some(JsValue::Int(_mode)) = arg.get(2) {
                 let fdflag = if flag & 128 == 128 {
-                    wasi::FDFLAGS_NONBLOCK
+                    wasi_fs::FDFLAGS_NONBLOCK
                 } else {
-                    wasi::FDFLAGS_SYNC
+                    wasi_fs::FDFLAGS_SYNC
                 } | if flag & 8 == 8 {
-                    wasi::FDFLAGS_APPEND
+                    wasi_fs::FDFLAGS_APPEND
                 } else {
                     0
                 };
                 let oflag = if flag & 512 == 512 {
-                    wasi::OFLAGS_CREAT
+                    wasi_fs::OFLAGS_CREAT
                 } else {
                     0
                 } | if flag & 2048 == 2048 {
-                    wasi::OFLAGS_EXCL
+                    wasi_fs::OFLAGS_EXCL
                 } else {
                     0
                 } | if flag & 1024 == 1024 {
-                    wasi::OFLAGS_TRUNC
+                    wasi_fs::OFLAGS_TRUNC
                 } else {
                     0
                 };
                 let right = if flag & 1 == 1 || flag & 2 == 2 {
-                    wasi::RIGHTS_FD_WRITE
-                        | wasi::RIGHTS_FD_ADVISE
-                        | wasi::RIGHTS_FD_ALLOCATE
-                        | wasi::RIGHTS_FD_DATASYNC
-                        | wasi::RIGHTS_FD_FDSTAT_SET_FLAGS
-                        | wasi::RIGHTS_FD_FILESTAT_SET_SIZE
-                        | wasi::RIGHTS_FD_FILESTAT_SET_TIMES
-                        | wasi::RIGHTS_FD_SYNC
-                        | wasi::RIGHTS_FD_WRITE
+                    wasi_fs::RIGHTS_FD_WRITE
+                        | wasi_fs::RIGHTS_FD_ADVISE
+                        | wasi_fs::RIGHTS_FD_ALLOCATE
+                        | wasi_fs::RIGHTS_FD_DATASYNC
+                        | wasi_fs::RIGHTS_FD_FDSTAT_SET_FLAGS
+                        | wasi_fs::RIGHTS_FD_FILESTAT_SET_SIZE
+                        | wasi_fs::RIGHTS_FD_FILESTAT_SET_TIMES
+                        | wasi_fs::RIGHTS_FD_SYNC
+                        | wasi_fs::RIGHTS_FD_WRITE
                 } else {
                     0
-                } | wasi::RIGHTS_FD_FILESTAT_GET
-                    | wasi::RIGHTS_FD_SEEK
-                    | wasi::RIGHTS_POLL_FD_READWRITE
-                    | wasi::RIGHTS_FD_READ
-                    | wasi::RIGHTS_FD_READDIR;
-                let res = unsafe { wasi::path_open(3, 0, path.as_str(), oflag, right, 0, fdflag) };
+                } | wasi_fs::RIGHTS_FD_FILESTAT_GET
+                    | wasi_fs::RIGHTS_FD_SEEK
+                    | wasi_fs::RIGHTS_POLL_FD_READWRITE
+                    | wasi_fs::RIGHTS_FD_READ
+                    | wasi_fs::RIGHTS_FD_READDIR;
+                let res =
+                    unsafe { wasi_fs::path_open(3, 0, path.as_str(), oflag, right, 0, fdflag) };

```

Comment:
the param `fd` in `path_xxx` is mean preopen dir fd.

if I preopen dir are `dir_a`  and `dir_b`. they `fd` are 3 and 4.
if I want open `dir_a/abc/text.txt`, it code is `wasi_fs::path_open(3,0,"abc/text.txt",oflag, right, 0, fdflag)`.
so the `fd` is not always `3`.

your can get the preopen dir by this way.(https://github.com/rust-lang/rust/blob/d8613f792c11d6d348b15eee79da561323fa0199/library/std/src/sys/wasi/fs.rs#L696)

maybe your need copy this function to `wasi_fs.rs`